coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,300 @@
+        'Maximum number of character for a line.',
+        (79, 80), default=80)
+
+
+@Length.subaspect
+class FileLength:
+    """
+    This aspect checks for files with a large number of lines in your codebase.
+    """
+    class docs:
+        example = """
+        # This file would be a large file if we assume that the max number of
+        # lines per file is 10
+
+        class node:
+            def __init__(self, value, leftmostchild, leftsibling):
+                self.value=value
+                self.leftmostchild=leftmostchild
+                self.leftsibling=leftsibling
+
+        # This is example is just showing what this aspect is about, because the
+        # max number of lines per file is usually 999.
+        """
+        example_language = 'python'
+        importance_reason = """
+        Too long programs (or files) are difficult to read, maintain and
+        understand.
+        """
+        fix_suggestions = """
+        Spliting files into modules, writing shorter methods and classes.
+        """
+    max_file_length = Taste[int](
+        'Maximum number of line for a file',
+        (999,), default=999)
+
+
+@Formatting.subaspect
+class Spacing:
+    """
+    This aspect checks on the spacing in your codebase (trailing
+    spaces, empty lines, indentation etc...)
+    """
+    class docs:
+        example = """
+        # Here is an example of code with spacing issues including trailing
+        # spaces, optional/useless blank lines and missing space aroud
+        # operators.
+
+
+
+        def func(   ):
+            return        37*-+2
+        """
+        example_language = 'python'
+        importance_reason = """
+        Useless spacing affects the readability and maintainability of your
+        code.
+        """
+        fix_suggestions = """
+        Removing the trailing spaces and the meaningless blank lines.
+        """
+
+
+@Spacing.subaspect
+class TrailingSpace:
+    """
+    This aspect checks for trailing spaces in your code.
+    """
+    class docs:
+        example = """
+        # Here is an example of code with trailing spaces.
+
+        def func(  a   ):
+            pass
+        """
+        example_language = 'python'
+        importance_reason = """
+        Trailing spaces make code less readablen and maintainaible.
+        """
+        fix_suggestions = """
+        Removing the trailing spaces.
+        """
+    allow_traling_spaces = Taste[bool](
+        'Determines whether or not traling spaces should be allowed or not.',
+        (True, False), default=False)
+
+
+@Spacing.subaspect
+class BlankLine:
+    """
+    This aspect checks for blank lines in your codebase.
+    """
+    class docs:
+        example = """
+        # There are two blank lines in this file, and the help improving
+        # on its readability by seperating imports from the rest of the code.
+        from coalib.bearlib.aspects import Root
+
+
+        @Root.subaspect
```

Comment:
don't use aspect code, it's too complicated. I'd rather go for hello world examples from wikipedia. Generally!